### PR TITLE
Fix error pages and redirects

### LINKED ISSUE #208: Search Google for landscape on premises, returns 500 link
## Summary

Search Google for landscape on premises
Get https://docs.ubuntu.com/landscape/en/onprem-overview as one of results
get 500

## Current and expected result

Expect old pages to not be  in Google..

### PR DESCRIPTION
**Pull request by nottrobin**
This does a bunch of things to fix #208 and some Django v2 upgrade oversights:

- Redirect `/landscape/en/onprem-overview` to `/landscape/en/onprem`
- Remove broken custom logic for error pages
- Tidy up templates for 404 and 500 pages a bit
- Upgrade whitenoise to v4
- Replace unslashed with removing slash in `redirects.yaml`
- Set `DJANGO_DEBUG` properly

QA
--

`./run clean && ./run` (this might take a little while)

- Go to http://0.0.0.0:8007/landscape/en/onprem-overview, check you're redirected to http://0.0.0.0:8007/landscape/en/onprem.
- Go to http://0.0.0.0:8007/landscape/en/onprem/ or http://0.0.0.0:8007/landscape/en/onprem.html, check you're redirected to http://0.0.0.0:8007/landscape/en/onprem
- Go to http://0.0.0.0:8007/landscape/en/onprem-erroneous, check you see a Django debug page about a 404

Now kill the server. Open up `webapp/urls.py` and add an "error" path lines:

``` python3
from django.http import HttpResponseServerError
...
urlpatterns += [
    url("error", lambda request: HttpResponseServerError()),
    url(r"^(?P<template>.*)/?$", TemplateFinder.as_view())
]
```

Now run `./run --env DJANGO_DEBUG=false`, and:

- Go to http://0.0.0.0:8007/landscape/en/onprem-erroneous - check you see a relatively nice 404 page
- Go to http://0.0.0.0:8007/error - check you see a relatively nice 500 server error page
- Browse the site generally, check everything works (e.g. http://0.0.0.0:8007/, http://0.0.0.0:8007/landscape, http://0.0.0.0:8007/landscape/en)

### CODE REVIEW

File: templates/404.html [hunk 1]
```diff
@@ -5,15 +5,10 @@
     <div class="row col-8">
         <h1>Ubuntu documentation</h1>
         <h2>Page not found (404)</h2>
-        <p>There is no page at this URL. How would you like to proceed?</p>
+        <p>To find what you're looking for you could try:</p>
         <ul>
-            {% if '/maas/' in request_path %}
```
Comment (nottrobin) [inline templates/404.html]: We no longer need special messages for maas or core, as maas is redirected straight over to docs.maas.io, and core is a separate application in k8s - its 404s are handled by that app.

File: templates/500.html [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
+        <h2>Server error (500)</h2>
+        <p>We encountered an error trying to serve this page. If you have time, please <a href="https://github.com/canonical-websites/docs.ubuntu.com/issues/new">file an issue</a> about the problem.</p>
+        <p>To find what you're looking for you could try:</p>
+        <ul>
+            <li>Viewing <a href="/">all documentation</a></li>
+            <li><a href="https://www.ubuntu.com/search">Searching all Ubuntu sites</a></li>
+        </ul>
+    </div>
+    <div class="row col-4">
+        <img src="https://manager.assets.ubuntu.com/static/hex-what.svg" />
```
Comment (tbille) [inline templates/500.html]: https://assets.ubuntu.com/v1/bcdcf2c8-image-404.svg

File: templates/404.html (hunk 1 above)
Comment (tbille) [reply templates/404.html]: shouldn't we have a redirect instead?

Those pages are indexed on search engines.

Comment (nottrobin) [reply templates/404.html]: It does redirect. Did you try going to http://docs.ubuntu.com/maas/nonexistentpage? Doesn't it redirect for you?

Comment (tbille) [reply templates/404.html]: oh sorry I missread your comment.

**Pull request merged**